### PR TITLE
Where does no longer accept array for OR conditions

### PR DESCRIPTION
**Pull request by mvorisek**
help wanted:
- doc must be update
- atk4/data must be updated, this test is failing:
  ```
  atk4\data\tests\ConditionSqlTest::testOrConditions
  ```

### CODE REVIEW

**Comment by DarkSide666** [comment]
Why remove this functionality? Does it interferes with something?

**Comment by mvorisek** [comment]
@DarkSide666 it is hack, we need to be sure we actually reject array/wrong input instead of threating it differently

As this method is more or less used only from data Scopes, we can easily make sure we do not generate this input

In the future, we may also introduce another junctions much more easiyl

@georgehristov promised to update the Scopes to not use this

**Comment by mvorisek** [comment]
@georgehristov ok with you?

**Comment by mvorisek** [comment]
@georgehristov can you please help with the docs for this?

**Comment by mvorisek** [comment]
@georgehristov can you adjust data Scopes so we can merge this, eg. to not rely on array nesting level?

**Comment by georgehristov** [comment]
@mvorisek so what is the solution for handling OR conditions now?
```
$q->where($q->orExpr()->where('a',1)->where('b',1));
```
Is it the one above? You have deleted the code comments for it

**Comment by mvorisek** [comment]
yes, passing something like `$q->orExpr()` directly

**Comment by georgehristov** [comment]
Prima-vista `data` does not rely on this "or from array" functionality but I am checking now

**Comment by georgehristov** [comment]
Pls merge develop into this branch...

**Comment by mvorisek** [comment]
rebased on develop

**Comment by georgehristov** [comment]
Thanks. I just ran the tests on `data` using this branch and all is fine?

File: src/Query.php [hunk 1]
```diff
@@ -534,27 +534,7 @@ public function _render_join()
...
-     *
-     *  $q->where([
-     *      ['age',20],
-     *      'age is null'
-     *  ]);
-     *
-     * The above use of OR conditions rely on orExpr() functionality. See
-     * that method for more information.
-     *
-     * To specify OR conditions
```
Comment (georgehristov) [inline src/Query.php:554]: I would keep this below comment here

**Review by georgehristov — APPROVED** [review]

File: src/Query.php (hunk 1 above)
Comment (mvorisek) [reply src/Query.php:554]: 👌 , comment reverted

File: src/Query.php [hunk 2]
```diff
@@ -571,17 +554,9 @@ public function where($field, $cond = null, $value = null, $kind = 'where', $num
...
-                if (is_array($row)) {
-                    $or->where(...$row);
-                } else {
-                    $or->where($row);
-                }
-            }
-            $field = $or;
+        // remove in v2.5
+        if (is_array($field)) {
+            throw new Exception('Array input / OR conditions is no longer supported');
```
Comment (DarkSide666) [inline src/Query.php:559]: This was easy and handy method how to define OR conditions :(

Comment (mvorisek) [reply src/Query.php:559]: and how to possibly and silently pass mistakes

**Pull request merged**